### PR TITLE
Enable optional threading for LHCoptim

### PR DESCRIPTION
**Pull request by thompsonmj**

### CODE REVIEW

File: src/QuasiMonteCarlo.jl [hunk 1]
```diff
@@ -40,7 +40,9 @@ struct LatinHypercubeSample <: SamplingAlgorithm end
 
 Samples using a Latin Hypercube.
 """
-struct LatinHypercubeSample <: SamplingAlgorithm end
+@with_kw struct LatinHypercubeSample{T} <: SamplingAlgorithm
```
Comment (ChrisRackauckas) [inline src/QuasiMonteCarlo.jl:43]: This is built into Base these days IIRC?

Comment (thompsonmj) [reply src/QuasiMonteCarlo.jl:43]: 🤷‍♂️

Comment (ChrisRackauckas) [reply src/QuasiMonteCarlo.jl:43]: `Base.@kwdef`

Comment (thompsonmj) [reply src/QuasiMonteCarlo.jl:43]: I can only see how to get this type of functionality with Parameters.

Comment (ChrisRackauckas) [reply src/QuasiMonteCarlo.jl:43]: What's the issue with `Base.@kwdef`?

Comment (thompsonmj) [reply src/QuasiMonteCarlo.jl]: A dangling comma after I got rid of Parameters was the issue, but `Base.@kwdef` works!

File: src/QuasiMonteCarlo.jl [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module QuasiMonteCarlo
 
-using Sobol, LatinHypercubeSampling, LatticeRules, Distributions
+using Sobol, LatinHypercubeSampling, LatticeRules, Distributions, Parameters
```
Comment (ChrisRackauckas) [inline src/QuasiMonteCarlo.jl:3]:
```suggestion
using Sobol, LatinHypercubeSampling, LatticeRules, Distributions
```

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Thanks!

File: src/QuasiMonteCarlo.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/QuasiMonteCarlo.jl]: Awesome. Thanks for the contribution.

**Pull request merged**